src/lib/decodeUnverifiedJWT.ts

```diff
@@ -0,0 +1,10 @@
+import jwt from "jwt-simple"
+
+export const decodeUnverifiedJWT = (token: string) => {
+  if (!token) {
+    return null
+  }
+
+  // NOTE: We don't verify the token when decoding here
+  return jwt.decode(token, "", true, "HS256")
```

Comment:
This makes sense to me. So MP doesn't verify the signed token, and blindly pass that that to upstream service (i.e. the authority) that will verify.